Allow user to disable extension for host

beszel opened the pull request:
Allows the user to turn off the extension for a host using either the popup or the new options page.
This is an early PR and is still missing end-to-end tests for the popup interface, which will be added before this is resolved. These features are also minimally styled, and the style will not be finished as part of this PR.

To test these changes, experiment with the popup and the options page. The options page can be accessed from the DGtW extension page:
chrome://extensions/?options=kgeehecadkggegiegoamiabpdjpgjkhg
I can add more detailed manual testing instructions if appropriate.

Resolves #5, resolves #118.

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Add popup that redundantly displays header text

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Reproduce all DGtW header functionality in popup

The "Dismiss this header" button is not included in the popup, for
obvious reasons. The "Restore original content" button is fully working.
The message remains the same after restoring content.

beszel committed:
Apply style to popup buttons

beszel committed:
Apply style to popup buttons

beszel committed:
Run full Puppeteer #101

Also fixed "Restore original content" button in popup.

beszel committed:
Run full Puppeteer #101

Also fixed "Restore original content" button in popup.

beszel committed:
Add e2e test for popup restore button

This uses a hardcoded popup URL, which should be changed.

beszel committed:
Add e2e test for popup restore button

This uses a hardcoded popup URL, which should be changed.

beszel committed:
Incomplete e2e tests for popup

beszel committed:
Incomplete e2e tests for popup

beszel committed:
Update tests to use popup

The extension ID is also static now.

beszel committed:
Update tests to use popup

The extension ID is also static now.

beszel committed:
Change "show changes" to a checkbox

With appropriate end-to-end tests

beszel committed:
Change "show changes" to a checkbox

With appropriate end-to-end tests

beszel committed:
Rewrite extension status request

When the popup requests the extension status from the content script,
it now receives a keyword telling it which message to display. It
previously received the entire message that it should display.

The Status enum can't be imported from `popup.js`, so a temporary
solution is used.

beszel committed:
Rewrite extension status request

When the popup requests the extension status from the content script,
it now receives a keyword telling it which message to display. It
previously received the entire message that it should display.

The Status enum can't be imported from `popup.js`, so a temporary
solution is used.

beszel committed:
Replace toggle buttons with checkboxes

beszel committed:
Replace toggle buttons with checkboxes

beszel committed:
Show the restore button when appropriate

beszel committed:
Show the restore button when appropriate

beszel committed:
Fix status enum

beszel committed:
Fix status enum

beszel committed:
Move popup style to `popup.css`

beszel committed:
Move popup style to `popup.css`

beszel committed:
Remove header

beszel committed:
Remove header

beszel committed:
Remove tests for excluded domain functions

Remove tests for `getExcludedDomain` and `getWhyExcluded`

beszel committed:
Remove tests for excluded domain functions

Remove tests for `getExcludedDomain` and `getWhyExcluded`

beszel committed:
Revert "Remove tests for excluded domain functions"

This reverts commit 2694fc24397064ea67772958dff40e5761fecd27.

The removed tests should be left in, as the functions they are testing
will be put back in. Tests will not pass on this commit.

beszel committed:
Revert "Remove tests for excluded domain functions"

This reverts commit 2694fc24397064ea67772958dff40e5761fecd27.

The removed tests should be left in, as the functions they are testing
will be put back in. Tests will not pass on this commit.

beszel committed:
Show reason that extension is disabled on a site

If the extension had to be disabled on a particular site, the header
would explain why. Now the popup explains why.

beszel committed:
Show reason that extension is disabled on a site

If the extension had to be disabled on a particular site, the header
would explain why. Now the popup explains why.

beszel committed:
Explain in popup when the content is restored

A broken test was found and fixed as part of this commit.

beszel committed:
Explain in popup when the content is restored

A broken test was found and fixed as part of this commit.

beszel committed:
Clean up IDs and selectors

beszel committed:
Clean up IDs and selectors

beszel committed:
Update version number and remove TODOs

beszel committed:
Update version number and remove TODOs

beszel committed:
Add space between status message and controls

beszel committed:
Add space between status message and controls

beszel committed:
Merge branch 'popup' of github.com:glam-lab/degender-the-web into popup

beszel committed:
Add reload button to un-restore content

beszel committed:
Make end-to-end tests more consistent and thorough

beszel committed:
Remove leftover header CSS

beszel committed:
Add test options page

beszel committed:
Perform replacements only when not blacklisted

The extension doesn't load when the domain is blacklisted. This causes
an error when the extension icon is clicked.

The domain matching pattern is currently very simple and needs
improvement.

beszel committed:
Fix async storage access

beszel committed:
Clear empty strings from blacklist

beszel committed:
Add tests using a real web page

Local files don't have a normal URL scheme, and I think it's important
to test a page with a typical URL.

beszel committed:
Fix tests and rename `blacklist` to `denyList`

beszel committed:
Merge branch 'master' into options

beszel committed:
Put `setDenyList` function in bootstrap

Following dry principle.

beszel committed:
Remove TODOs

beszel committed:
Popup explains if DGtW is turned off on the page

beszel committed:
Popup tests for when DGtW is turned off on a page

beszel committed:
Change denyList to doNotReplaceList

beszel committed:
Replace options list textarea with new prototype

The textarea is replaced with an unordered list and a text field for
adding new items. The options page is unstyled.

beszel committed:
Refactor options for readability

beszel committed:
Use enter key to add to the doNotReplaceList

beszel committed:
Minimally style options page

beszel committed:
Show "turned off" label conditionally

beszel committed:
Make doNotReplaceList ignore protocol

beszel committed:
Fix indentation of popup.html for consistency

beszel committed:
Refactor hiding and showing popup elements

beszel committed:
Improve popup doNotReplaceList interactions

The doNotReplaceList now supports only hosts, not longer URLs.

beszel committed:
Fix reload button visibility fault

Also refactored `showElement` and `hideElement` functions into
`setVisibility` to make this possible.

beszel committed:
Update existing tests for new options page

beszel committed:
Rename end-to-end tests for options

beszel committed:
Improve doNotReplaceList

This commit is bigger than it should be! It includes the following:

- New and updated tests
- A much better way to test the options page correctly
- Removed redundant "Restore original content" button
- All tests are now passing

beszel committed:
DRY refactor of `storage.js`

Not quite as DRY as it could be, but it's become a time sink at this
point.

beszel committed:
Add tests for new options page

beszel committed:
Fix test message

ProfJanetDavis commented on src/content.js:
```diff
@@ -1,8 +1,27 @@
-    contentScript.main();
+
+    // Omit inessential URL components when performing comparison.
+    const url = location.host + location.pathname;
```
Does it make sense to declare/call a function to ensure we handle urls consistently across the codebase?

beszel replied:
This is the only place in the code where we have access to `location`, and the `host` and `pathname` properties are the best way of getting what we need. Other cases include user provided URLs and end-to-end testing, both of which must be handled quite differently.

ProfJanetDavis commented on src/content.js:
```diff
@@ -1,8 +1,27 @@
+        function(items) {
+            // If DGtW is allowed to access this page, run the content script.
+            if (
+                // Check if the URL matches one of the items in the list.
```
// Check if the URL matches NONE of the items in the list

ProfJanetDavis replied:
(That ! is easy to miss if you are reading quickly)

ProfJanetDavis commented on src/options.js:
```diff
@@ -0,0 +1,119 @@
+            .join("://");
+    }
+
+    url = url.split("/")[0];
```
Use URL manipulation functions instead of string matching.

beszel replied:
The URL API requires a protocol, and since URLs in the list don't have protocols, we want to let the user submit a URL that doesn't have a protocol. I've added a comment line mentioning why.

ProfJanetDavis commented on src/options.js:
```diff
@@ -0,0 +1,119 @@
+            return s === urlToRemove;
+        });
+
+        // If the item hasn't already been removed from the list,
```
This comment is an incomplete thought

ProfJanetDavis commented on src/options.js:
```diff
@@ -0,0 +1,119 @@
+        saveDoNotReplaceList(doNotReplaceList);
+
+        // Update the displayed list
+        displayDoNotReplaceList(doNotReplaceList);
```
Should this be inside the conditional?

beszel replied:
Putting `saveDoNotReplaceList` in the conditional makes sense; we don't need to overwrite the list if we haven't changed it. We'll still update the displayed list either way because we want to keep it synced with the list in `chrome.storage` whenever possible.

ProfJanetDavis commented on src/popup.js:
```diff
@@ -2,129 +2,218 @@
     chrome.tabs.query(queryInfo, function(tabs) {
         // Normally, only one tab matches the query.
         // When testing, a blank tab might match the query. Using `length - 1`
         // picks the correct target, the most recently opened background tab.
```
Thanks, this explanation is very necessary

ProfJanetDavis commented on src/popup.js:
```diff
@@ -2,129 +2,218 @@
-    document.getElementById(id).classList.remove("show");
-    document.getElementById(id).classList.add("hide");
+function setVisibility(id, visibility) {
+    if (visibility) {
```
if(isVisible) {

ProfJanetDavis replied:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

beszel replied:
I like the article! The point of this function is that something will be set to visible or not visible. I wouldn't feel good about `isVisible` because of the implication that it's the current state of the element. I've chosen `shouldBeVisible`, since "is" is a conjugation of "to be". It also adequately implies that setting it to `false` hides the element.

ProfJanetDavis commented on src/popup.js:
```diff
@@ -139,22 +228,82 @@ function getUrlParameter(sParam) {
+    // If a status has made the reload button visible, we can't hide it here
+    const reloadVisibility = document
+        .getElementById(ids.reloadPage)
+        .classList.contains("show");
```
Is `reloadVisibility` a boolean?

beszel replied:
Yes, I've renamed it clumsily to `doShowReloadButton`. Any other ideas? Is just `showReloadButton` clear or does that look like a function?

ProfJanetDavis commented on src/popup.js:
```diff
@@ -139,22 +228,82 @@ function getUrlParameter(sParam) {
+    document.getElementById(ids.reloadMessage).innerHTML = message;
+}
+
+function toggleHostWithoutScript() {
```
Needs introductory documentation. 

Is this function a callback? If so, use that naming convention (or don't) consistently.

beszel replied:
I found mention of `Callback` and `Handler` suffixes, and I chose to use `Handler` since it responds to a click from the user. This change brings the risk of sounding as though the method handles scripts.

Its counterpart `toggleHostWithScript` needs documentation too, but it's called both as a handler and on its own. I haven't worked with callbacks much before this, so I'm not familiar with the conventions. What do you think makes sense here?

ProfJanetDavis commented on src/popup.js:
```diff
@@ -139,22 +228,82 @@ function getUrlParameter(sParam) {
+
+    // If a status has made the reload button visible, we can't hide it here
+    // FIXME This gets the checkedness *on click*, not when the popup is opened
+    //       This variable is no longer being used, since it doesn't work right
```
If a variable is not being used, it should be removed from the code.

ETA: I'm confused - it looks like the variable is being used below. Also, should it be called `isChecked?` What does "checked" mean in this context?

beszel replied:
`isChecked` makes sense, I was mirroring the `checked` property of the checkbox.
The `FIXME` part is my mistake! There was previously a `reloadVisibility` variable that didn't do what it should. I removed it and fixed the function but forgot to remove the comment. I've removed it now.

ProfJanetDavis commented on src/popup.js:
```diff
@@ -169,4 +318,7 @@ document
 // Display extension status when popup is opened
+// FIXME Since this is async, the popup is supposedly loaded before its status.
+//       Fixing this should make tests simpler and require less waiting for
+//       element visibility.
```
How can this `FIXME` be resolved?

beszel replied:
After this comment was made, you and I discussed this in person. I've removed the `FIXME` message because the supposed fault only affects tests and we can work around it very easily.

ProfJanetDavis commented on test/e2e/content/case.turned-off-for-host.test.js:
```diff
@@ -0,0 +1,141 @@
+            });
+        });
+    });
+
```
Correct the test descriptions for the three tests above

ProfJanetDavis replied:
(and make sure the tests are correct!)

beszel replied:
Got it! I consistently use `visible` instead of `hidden` now too.

ProfJanetDavis commented on test/e2e/content/case.turned-off-for-host.test.js:
```diff
@@ -0,0 +1,141 @@
+            await popup.waitForSelector(selectors.reloadPage, { hidden: true });
+        });
+    });
+});
```
Missing test: The "allow replacements on host" checkbox should be unchecked.

beszel replied:
I've now added the corresponding `turned-on` test as well.

ProfJanetDavis commented on test/e2e/content/case.turned-on-for-host.test.js:
```diff
@@ -0,0 +1,33 @@
+            "They washed their motorcycle. They washed their car."
+        );
+    });
+});
```
This looks incomplete

beszel replied:
I've added more now, I believe these tests are now complete.

ProfJanetDavis commented on test/e2e/interaction/options.test.js:
```diff
@@ -0,0 +1,129 @@
+            const hosts = await getListedHosts(options);
+            expect(hosts).to.have.lengthOf(4);
+            expect(hosts).to.include("test.edu");
+        });
```
Should there be two separate tests: one for what appears in the UI and one for what appears in storage?

ProfJanetDavis replied:
Is this useful, or redundant to earlier tests?

beszel replied:
Makes sense.

ProfJanetDavis commented on test/e2e/interaction/options.test.js:
```diff
@@ -0,0 +1,129 @@
+        it.only("should clear the input field", async function() {
+            expect(await options.$eval("#newEntry", e => e.value)).to.equal("");
+        });
+    });
```
Remove "only" from this test and the one above

ProfJanetDavis commented on test/e2e/storage.js:
```diff
@@ -0,0 +1,54 @@
+/*globals chrome */
+// _pageBindings solution from https://stackoverflow.com/a/59138297/9158894
```
Explain why this module exists.

beszel committed:
Improve comments and identifiers

beszel committed:
Improve storage module to avoid `exposeFunction`

This simpler solution for the storage module avoids the `_pageBindings`
hack, which seemed to be causing some problems with multiple calls to
exposed functions with the same name. The storage method signatures are
not changed.

beszel committed:
Fix `genericReload` status message not appearing

beszel committed:
Minor: Use `console.warn` for options page warning

beszel committed:
Remove `FIXME` comment for popup asynchronicity

It isn't a big problem that the popup loads before its contents do. We
can wait for particular elements to appear if we need to test them.

beszel committed:
Add new tests for popup

beszel committed:
Remove leftover log message

beszel committed:
Add more popup tests and address PR suggestions

ProfJanetDavis commented on src/options.js:
```diff
@@ -0,0 +1,118 @@
+function addItem() {
+    const newEntryField = document.getElementById("newEntry");
+
+    // We can't use the URL API here, as it requires a protocol
```
Why would there not be a protocol? I think it's because the user might not type one, but I'm not totally sure.

ProfJanetDavis reviewed (changes requested):
In addition to one comment I want clarified, I see four failing e2e tests. I wonder if you have uncommitted code. Could you investigate and let me know?

If these failures appear related to the timeout problem you already told me about, I might want you to walk me through that. Thanks!

> 1) When the page includes the stopword 'gender', it
       should explain in the popup:
     AssertionError: expected '' to include 'gender'
      at Context.<anonymous> (test/e2e/content/case.mentions-gender.test.js:52:31)
      at processTicksAndRejections (internal/process/task_queues.js:85:5)

>  2) When the page includes the stopword 'gender', it
       should show the 'Show highlights' checkbox:
     Error: Timeout of 3000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/janet/chrome-extensions/degender-the-web/test/e2e/content/case.mentions-gender.test.js)
      at listOnTimeout (internal/timers.js:531:17)
      at processTimers (internal/timers.js:475:7)

>  3) When the page includes the stopword 'gender', it
       When the user checks the 'Show highlights' box
         "before all" hook for "should show them all":
     Error: Node is either not visible or not an HTMLElement
      at ElementHandle._clickablePoint (node_modules/puppeteer/lib/JSHandle.js:199:13)
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
      at async ElementHandle.click (node_modules/puppeteer/lib/JSHandle.js:265:20)
      at async DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:367:5)
      at async Context.<anonymous> (test/e2e/content/case.mentions-gender.test.js:98:13)
    -- ASYNC --
      at ElementHandle.<anonymous> (node_modules/puppeteer/lib/helper.js:111:15)
      at DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:367:18)
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
      at async Context.<anonymous> (test/e2e/content/case.mentions-gender.test.js:98:13)
    -- ASYNC --
      at Frame.<anonymous> (node_modules/puppeteer/lib/helper.js:111:15)
      at Page.click (node_modules/puppeteer/lib/Page.js:1031:29)
      at Context.<anonymous> (test/e2e/content/case.mentions-gender.test.js:98:25)
      at processImmediate (internal/timers.js:439:21)

>  4) When the page includes the stopword 'gender', it
       When the user unchecks the 'Show highlights' box
         "before all" hook for "should hide them all":
     Error: Node is either not visible or not an HTMLElement
      at ElementHandle._clickablePoint (node_modules/puppeteer/lib/JSHandle.js:199:13)
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
      at async ElementHandle.click (node_modules/puppeteer/lib/JSHandle.js:265:20)
      at async DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:367:5)
      at async Context.<anonymous> (test/e2e/content/case.mentions-gender.test.js:109:13)
    -- ASYNC --
      at ElementHandle.<anonymous> (node_modules/puppeteer/lib/helper.js:111:15)
      at DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:367:18)
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
      at async Context.<anonymous> (test/e2e/content/case.mentions-gender.test.js:109:13)
    -- ASYNC --
      at Frame.<anonymous> (node_modules/puppeteer/lib/helper.js:111:15)
      at Page.click (node_modules/puppeteer/lib/Page.js:1031:29)
      at Context.<anonymous> (test/e2e/content/case.mentions-gender.test.js:109:25)
      at processImmediate (internal/timers.js:439:21)

beszel commented:
@ProfJanetDavis On the failing e2e tests:
I'm not able to reproduce those errors, and my working tree is clean. Some of these test failures look like they're caused by the test race condition, so I suspect they all are.
Have you tried running the e2e tests on the `master` branch and then immediately on the `options` branch, several times each? If you do, make sure to leave Chromium in the foreground so we can control for that too. I suspect that the tests fail only when your computer is working hard on something else, so this experiment helps control for that.
If the problem is that the tests run before the extension makes changes, we should be able to fix most or all of them just by waiting in the `before` block for a DGtW-added element to appear on the page. I'd start now but since I can't reproduce the test failures I wouldn't know if or when it worked.

ProfJanetDavis commented:
Thanks- I wondered if it was a race condition. My laptop is pretty old. Let me try some of the things you suggest and see if I can resolve it. 

Even if it might be helpful, I’m not available to meet at noon today - I’m getting a new nanny oriented and that’s the start of nap time. I’ll let you know the results if my experiments. 

--
Janet Davis
Sent from my "smart" phone, in haste, with auto-(in)correct

> On May 26, 2020, at 8:21 PM, Ian Hawkins <notifications@github.com> wrote:
> 
> ﻿
> @ProfJanetDavis On the failing e2e tests:
> I'm not able to reproduce those errors, and my working tree is clean. Some of these test failures look like they're caused by the test race condition, so I suspect they all are.
> Have you tried running the e2e tests on the master branch and then immediately on the options branch, several times each? If you do, make sure to leave Chromium in the foreground so we can control for that too. I suspect that the tests fail only when your computer is working hard on something else, so this experiment helps control for that.
> If the problem is that the tests run before the extension makes changes, we should be able to fix most or all of them just by waiting in the before block for a DGtW-added element to appear on the page. I'd start now but since I can't reproduce the test failures I wouldn't know if or when it worked.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

ProfJanetDavis commented:
@irhawk, you nailed it. If I keep Chrome in the foreground those e2e test
errors go away. However, there is one error that's persistent:

  1) When the extension is turned off for this host, the page
>
>        When the user has not yet reloaded the page, it
>
>          should not yet change the status message:
>
>
>       AssertionError: expected '' to equal 'Degender the Web has replaced
> gender pronouns on this page.'
>
>       + expected - actual
>
>
>       +Degender the Web has replaced gender pronouns on this page.
>
>
>
>       at Context.<anonymous>
> (test/e2e/content/case.turned-off-for-host.test.js:43:35)
>
>       at processTicksAndRejections (internal/process/task_queues.js:85:5)
>

Is it possible this is a race condition also? Do you want to investigate,
or give me a suggestion for where to start? Thanks!

On Wed, May 27, 2020 at 9:22 AM <davisj@whitman.edu> wrote:

> Thanks- I wondered if it was a race condition. My laptop is pretty old.
> Let me try some of the things you suggest and see if I can resolve it.
>
> Even if it might be helpful, I’m not available to meet at noon today - I’m
> getting a new nanny oriented and that’s the start of nap time. I’ll let you
> know the results if my experiments.
>
> --
> Janet Davis
> Sent from my "smart" phone, in haste, with auto-(in)correct
>
> On May 26, 2020, at 8:21 PM, Ian Hawkins <notifications@github.com> wrote:
>
> ﻿
>
> @ProfJanetDavis <https://github.com/ProfJanetDavis> On the failing e2e
> tests:
> I'm not able to reproduce those errors, and my working tree is clean. Some
> of these test failures look like they're caused by the test race condition,
> so I suspect they all are.
> Have you tried running the e2e tests on the master branch and then
> immediately on the options branch, several times each? If you do, make
> sure to leave Chromium in the foreground so we can control for that too. I
> suspect that the tests fail only when your computer is working hard on
> something else, so this experiment helps control for that.
> If the problem is that the tests run before the extension makes changes,
> we should be able to fix most or all of them just by waiting in the before
> block for a DGtW-added element to appear on the page. I'd start now but
> since I can't reproduce the test failures I wouldn't know if or when it
> worked.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/glam-lab/degender-the-web/pull/122#issuecomment-634404086>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEXJIUZSPP4A2WFF5UM2DFDRTSBNLANCNFSM4M5VX6MA>
> .
>
>

-- 
Janet Davis (she/her or they/them)
Associate Professor and Microsoft Chair of Computer Science, Whitman
College
509-527-5758, http://cs.whitman.edu/~davisj/
Schedule an appointment: http://meetme.so/JanetDavis

beszel commented:
@ProfJanetDavis I'm glad it helped! That looks like a race condition where some test needs to wait for the popup status message, I'll take a look tonight. I expect to have a code snippet or temporary branch for you to try.

beszel committed:
Hide popup status when empty

This lets some tests wait for the status message to appear.

beszel committed:
Add next step, commented out

Uncomment the `page.waitFor` line to check if there's a race condition
between the popup opening and the extension applying to the page.

beszel commented:
@ProfJanetDavis Check out the `options-experiment` branch and try running the e2e tests there. I made the change we discussed before where the popup message remains hidden until filled, and tests can wait for it to appear. The test passes for me, but it passed before this change too (unfortunately).

If this works for you, make sure to run the tests again on `options` right away to make sure a fluke didn't cause the tests to pass even without these changes.

As a next step, try uncommenting the `page.waitFor` line I added in the `before` block. It's not a real solution, but it will give us an idea of whether there's a race condition between the changes to Chrome storage and the extension applying to the page.

ProfJanetDavis commented:
Good advice. The error in the options branch does indeed seem to be
non-deterministic - those tests sometimes all passed. However,

   - The options-experiment branch only had test failures when I
   accidentally navigated away from the Puppeteer instance of Chrome so that
   it lost focus. Otherwise, every test passed.
   - The options branch had test failures over half the time, before and
   after running tests on the other branch.

I'm not sure which line of which file you are suggesting I uncomment, or in
which branch. I tried a few things and all resulted in multiple test
failures. Can you clarify? Or is that experiment moot with the success of
the one above?

On Wed, May 27, 2020 at 10:07 PM Ian Hawkins <notifications@github.com>
wrote:

> @ProfJanetDavis <https://github.com/ProfJanetDavis> Check out the
> options-experiment branch and try running the e2e tests there. I made the
> change we discussed before where the popup message remains hidden until
> filled, and tests can wait for it to appear. The test passes for me, but it
> passed before this change too (unfortunately).
>
> If this works for you, make sure to run the tests again on options right
> away to make sure a fluke didn't cause the tests to pass even without these
> changes.
>
> As a next step, try uncommenting the page.waitFor line I added in the
> before block. It's not a real solution, but it will give us an idea of
> whether there's a race condition between the changes to Chrome storage and
> the extension applying to the page.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/glam-lab/degender-the-web/pull/122#issuecomment-635103896>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEXJIU6NN2DQO3AI5GZ6JR3RTXWRHANCNFSM4M5VX6MA>
> .
>


-- 
Janet Davis (she/her or they/them)
Associate Professor and Microsoft Chair of Computer Science, Whitman
College
509-527-5758, http://cs.whitman.edu/~davisj/
Schedule an appointment: http://meetme.so/JanetDavis

beszel committed:
Clean up experiment for use

beszel committed:
Clarify explanation for not using URL API

beszel commented:
Perfect! If that works consistently, we don't need to try the second approach. I've cleaned up the experimental stuff and removed the commented-out "plan B" lines, then I merged it into the `options` branch. I also clarified the comment you pointed out earlier, let me know what you think.

ProfJanetDavis commented:
Merged to master and made a small formatting improvement to the popup. I
think we are all set. Thanks for all your work on this!

Let me know if you want me to write a LinkedIn testimonial - or if you want
to keep working on this as a personal project, for that matter.

Janet

On Thu, May 28, 2020 at 2:29 PM Ian Hawkins <notifications@github.com>
wrote:

> Perfect! If that works consistently, we don't need to try the second
> approach. I've cleaned up the experimental stuff and removed the
> commented-out "plan B" lines, then I merged it into the options branch. I
> also clarified the comment you pointed out earlier, let me know what you
> think.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/glam-lab/degender-the-web/pull/122#issuecomment-635618531>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEXJIU6JXFT3QQSSKQJOJFTRT3JU5ANCNFSM4M5VX6MA>
> .
>


-- 
Janet Davis (she/her or they/them)
Associate Professor and Microsoft Chair of Computer Science, Whitman
College
509-527-5758, http://cs.whitman.edu/~davisj/
Schedule an appointment: http://meetme.so/JanetDavis
